### ISSUE TITLE
Elegant retina display detection and handling

### ISSUE THREAD

**Issue by nbrochu** [SerpentAI/SerpentAI#37]
Right now on macOS, the captured frames are stored as 4X larger images (2X width, 2X height) on retina displays. You can still get a Game Agent working by resizing the image but this should be done at the frame grabber level.

I'm thinking a nice way of doing this would be to capture a single frame on FrameGrabber initialization and look at if the resolution matches the requested geometry. If not, we can activate a flag that will resize the image data in that FrameGrabber before it ever gets to the agent.

**Comment by d0p3t** [comment]
Checking for a single frame in FrameGrabber initialisation works, but there is an issue afterwards. While the game_agent is running, it continuously grabs frames using the class method `FrameGrabber.get_frames()`. 

This is fine, but the game calls this method using the following code snippet:

```python
    @offshoot.forbidden
    def grab_latest_frame(self):
        game_frame_buffer = FrameGrabber.get_frames(
            [0],
            (
                self.window_geometry.get("height"),
                self.window_geometry.get("width"),
                3
            )
        )

        return game_frame_buffer.frames[0]
```

In other words, it's getting the `window_geometry` from the `window controller` and grabbing frames according to that dimension, hence checking for a single frame at FrameGrabber initialisation does not work, unless there is a way to get a flag that we set in initialisation and pass it through that `get_frames()` method. As far as I know, this require instantiation of an `cls` object which is far from ideal.

Perhaps someone with more python experience knows where to set a flag that can be used in `FrameGrabber.get_frames()`

Code I used to resize width/height on FrameGrabber initialisation (`frame_grabber.py`)

```python
self.frame_check = self.grab_frame()

if self.frame_check.shape[0] == (self.height / 2) and self.frame_check.shape[1] == (self.width / 2):
    self.width = width / 2
    self.height = height /2
```
